braindecode/models/modules.py

```diff
@@ -523,3 +529,331 @@ def forward(self, x):
+        self.band_filters = band_filters
+        self.n_bands = len(band_filters)
+        self.phase = phase
+        self.method = method
+        self.n_chans = n_chans
+
+        self.method_iir = True if self.method == "iir" else False
+
+        if self.method_iir:
+            if iir_params is None:
+                iir_params = dict(output="ba")
+            else:
+                if "output" in iir_params:
+                    if iir_params["output"] == "sos":
+                        warn(
+                            "It is not possible to use second-order section filtering with Torch. Changing to filter ba",
+                            UserWarning,
+                        )
+                        iir_params["output"] = "ba"
+
+        self._apply_filter_func = None
+
+        filts = {}
+        for idx, (l_freq, h_freq) in enumerate(band_filters):
+            filt = create_filter(
+                data=None,
+                sfreq=sfreq,
+                l_freq=l_freq,
+                h_freq=h_freq,
+                filter_length=filter_length,
+                l_trans_bandwidth=l_trans_bandwidth,
+                h_trans_bandwidth=h_trans_bandwidth,
+                method=method,
+                iir_params=iir_params,
+                phase=phase,
+                fir_window=fir_window,
+                fir_design=fir_design,
+                verbose=verbose,
+            )
+            if not self.method_iir:
+                b = from_numpy(filt).float()
+                filts[f"band_{idx}"] = {"b": b}
+
+            else:
+                _check_coefficients((filt["b"], filt["a"]))
+                b = torch.tensor(filt["b"], dtype=torch.float64)
+                a = torch.tensor(filt["a"], dtype=torch.float64)
+
+                filts[f"band_{idx}"] = {"b": b, "a": a}
+
+        self.filts = nn.ParameterDict(filts)
+
+        if self.method_iir:
+            self._apply_filter_func = self._apply_iir
+        else:
+            self._apply_filter_func = partial(self._apply_fir, n_chans=self.n_chans)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """
+        Apply the filter bank to the input signal.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            Input tensor of shape (batch_size, time_points).
+        n_chans: int
+            Number of chans passed internally inside the object
+
+        Returns
+        -------
+        torch.Tensor
+            Filtered output tensor of shape (batch_size, n_bands, n_chans, filtered_time_points).
+        """
+        # Initialize a list to collect filtered outputs
+        n_bands = self.n_bands
+        output = []
+
+        for band_idx in range(n_bands):
+            filtered = self._apply_filter_func(x, self.filts[f"band_{band_idx}"])  # type: ignore
+            output.append(filtered)
+
+        # Shape: (batch_size, n_bands, nchans, time_points)
+        output = torch.cat(output, dim=1)
+        return output
```

Comment:
Isn't it faster to do:
```suggestion
        return torch.cat(
            self._apply_filter_func(x, p_filt) for p_filt in self.filts.values(),
            dim=1
         )
```